x-pack/plugins/security_solution/server/lib/detection_engine/signals/wrap_hits_factory.ts

```diff
@@ -25,11 +25,15 @@ export const wrapHitsFactory = ({
   const wrappedDocs: WrappedSignalHit[] = events.flatMap((doc) => [
     {
       _index: signalsIndex,
-      // TODO: bring back doc._version
-      _id: generateId(doc._index, doc._id, '', ruleSO.attributes.params.ruleId ?? ''),
+      _id: generateId(
+        doc._index,
+        doc._id,
+        doc._version ? doc._version.toString() : '',
```

Comment:
fwiw a lot of people use the constructor variant such as `String(doc._version)` because it's safer if someone changes this code and accidentally pushes in a non string.

One other thing is that I do not think the `doc._version` is the best usage of identifying unique document changes anymore within elastic.

In Saved Objects and other places if we are trying to identify if a document has changed and key off of that it's better to use `_seq_no and _primary_term`

Ref: https://www.elastic.co/guide/en/elasticsearch/reference/current/optimistic-concurrency-control.html
 
The caveat being we have to ensure we are passing the correct flags to return these two fields with our search results which you would have to check. You could fall back on `doc._version` if you do not see them being set to double check things or write an e2e test to ensure we do return them as expected as well at some point.

However, overall I think we have older deprecated patterns of using `doc._version` instead of using those two other fields and should prefer using those.

Let me know if that doesn't sound right.

Reply:
just pushed some changes and updated it as `String(doc._version)` for this moment. It looks like we would need to use the `seq_no_primary_term` flag, I haven't found out exactly how just yet, looking into it.

Reply:
created #102395